src/createElementFunction.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import Relay from 'react-relay';
+
+import * as Utils from './containerUtils';
+import QueryAggregator from './QueryAggregator';
+import getRouteParams from './getRouteParams';
+import RouteGenerator from './RouteGenerator';
+
+export function createElementFunction() {
+  const queryAggregator = new QueryAggregator();
+  const routeGenerator = new RouteGenerator();
+
+  return function createElement(Component, props) {
+    const {branch, route, params, location, routeParams} = props;
+
+    // Wrap any Relay.Containers
+    let element = !Relay.isContainer(Component) ? <Component {...props} /> :
+      Utils.createContainerElement(
+        Component,
+        props,
+        getRouteParams(routeParams, location.query, route.queryParams || []),
+        route.rootContainerProps || {},
+        routeGenerator.getRouteFor(branch),
+        queryAggregator
+      );
+
+    // Wrap the root component in a RootContainer
+    element = route !== branch[0] ? element :
```

Comment:
Another potential trap here: https://github.com/rackt/react-router/blob/v1.0.0-beta3/modules/Router.js#L270-L271

If there is no component specified for the root route, then we'll never hit the forward branch here.